### PR TITLE
New CVar `sv_rcon_allowexternal` for control external usage of RCon

### PR DESCRIPTION
**Pull request by GiovaniFerraroTrivelli**
Some panels, like OpenGamePanel, use an option to send commands through Rcon. In my case, for security, I disabled Rcon (rcon_password "") but I need an option to send commands without entering to the server.

I added a cvar, sv_rcon_allowexternal (0/1) that allows or disallows the usage of Rcon from externals IPs. Also, it logs when this cvar value is 0 and someone tries to use Rcon.

This is more simple than #587, which is a very good idea. In the future it could be improved to match this suggestion.

(This PR recreates correctly #710)

### CODE REVIEW

**Comment by afwn90cj93201nixr2e1re** [comment]
.

File: rehlds/engine/sv_main.cpp [hunk 1]
```diff
@@ -3376,6 +3379,15 @@ void SV_Rcon(netadr_t *net_from_)
 	Q_memcpy(remaining, &net_message.data[Q_strlen("rcon")], len);
 	remaining[len] = 0;
 
+	#ifdef REHLDS_FIXES
+	if(!sv_rcon_allowexternal.value && !NET_CompareBaseAdr(*net_from_, net_local_adr))
+	{
+		Con_Printf("Rcon denied from %s\n", NET_AdrToString(*net_from_));
```
Comment (theAsmodai) [inline rehlds/engine/sv_main.cpp:3385]: Please hide it under sv_rcon_condebug.value check

Comment (GiovaniFerraroTrivelli) [reply rehlds/engine/sv_main.cpp]: Done

**Comment by afwn90cj93201nixr2e1re** [comment]
.

**Comment by afwn90cj93201nixr2e1re** [comment]
.

**Comment by afwn90cj93201nixr2e1re** [comment]
.

**Comment by GiovaniFerraroTrivelli** [comment]
Conflict resolved

**Comment by afwn90cj93201nixr2e1re** [comment]
.

**Comment by afwn90cj93201nixr2e1re** [comment]
.

**Review by wopox1337 — DISMISSED** [review]

**Comment by wopox1337** [comment]
> This is more simple than #587

I do not agree with you.

The problem you described refers to problems with game server control panels (like OGP). The control panel should use stdin for local control.

**Remote** CONsole = (RCon). 

The proposed functionality is a "crutch" for a remote control method called RCon.

File: rehlds/engine/sv_main.cpp [hunk 2]
```diff
@@ -3416,6 +3419,18 @@ void SV_Rcon(netadr_t *net_from_)
 	Q_memcpy(remaining, &net_message.data[Q_strlen("rcon")], len);
 	remaining[len] = 0;
 
+#ifdef REHLDS_FIXES
+	if(!sv_rcon_allowexternal.value && !NET_CompareBaseAdr(*net_from_, net_local_adr))
```
Comment (wopox1337) [inline rehlds/engine/sv_main.cpp:3423]: `if(` -> `if (`

File: rehlds/engine/sv_main.cpp [hunk 3]
```diff
@@ -3416,6 +3419,18 @@ void SV_Rcon(netadr_t *net_from_)
 	Q_memcpy(remaining, &net_message.data[Q_strlen("rcon")], len);
 	remaining[len] = 0;
 
+#ifdef REHLDS_FIXES
+	if(!sv_rcon_allowexternal.value && !NET_CompareBaseAdr(*net_from_, net_local_adr))
+	{
+		if (sv_rcon_condebug.value > 0.0f)
+		{
+			Con_Printf("Rcon denied from %s\n", NET_AdrToString(*net_from_));
+			Log_Printf("Rcon denied from %s\n", NET_AdrToString(*net_from_));
```
Comment (wopox1337) [inline rehlds/engine/sv_main.cpp:3428]: Add better description

```cpp
Con_Printf("External Rcon denied from %s\n", NET_AdrToString(*net_from_));
Log_Printf("External Rcon denied from %s\n", NET_AdrToString(*net_from_));
```

**Review by wopox1337 — CHANGES_REQUESTED** [review]

File: rehlds/engine/sv_main.cpp (hunk 2 above)
Comment (wopox1337) [inline rehlds/engine/sv_main.cpp:3423]: `if(` -> `if (`

**Review by wopox1337 — CHANGES_REQUESTED** [review]

**Comment by Maxi605** [comment]
@theAsmodai can we get this?

**Review by wopox1337 — APPROVED** [review]

**Pull request closed without merging**